### ISSUE TITLE
Error on changing app service plan (previously reported)

### ISSUE THREAD

**Issue by jason-swissre** [hashicorp/terraform-provider-azurerm#14629]
**Community Note**

- Please vote on this issue by adding a 👍 reaction to the original issue to help the community and maintainers prioritize this request

- Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request

- If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Terraform (and AzureRM Provider) Version**
Terraform v1.0.7
provider.azurerm v2.87.0
**Affected Resource(s)**
azurerm_app_service
**Expected Behavior**
Resource re-created on changing app service plan as it used to happen before v1.24.0

**Actual Behavior**
azurerm_app_service.appservice: web.AppsClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: autorest/azure: Service returned an error. Status=<nil> <nil>

**Steps to Reproduce**
Change app_service_plan_id
terraform apply
**Important Factoids**

This a re-opening of case #3185.  That case was closed and #6348 opened to deal with the issue but #6348 was closed without any comment.  I am still having the issue with the versions listed above.

**Comment by jason-swissre** [comment]
Hello, has anyone been able to investigate this issue?

**Comment by CSymes** [comment]
Agreed - this is still an issue. #6348 was closed as a duplicate of itself (erroneously, I presume, while closing #3185), but should really be re-opened.

**Comment by katbyte** [comment]
It appears this issue relates to `azurerm_app_service`, which was removed in v5.0 in favour of `azurerm_linux_web_app` or `azurerm_windows_web_app` (see the [v5.0 upgrade guide](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/5.0-upgrade-guide#azurerm_app_service) for details), as such we are going to close this issue out as it no longer applies to the current version of the provider.

If this is still relevant on the latest version of the provider (v5.x) please open a new issue against `azurerm_linux_web_app` or `azurerm_windows_web_app` referencing this one. Thank you!
